### PR TITLE
[WIP] concrete_solve for NILSS 

### PR DESCRIPTION
**Pull request by frankschae**
It currently fails for me because of https://github.com/SciML/SciMLBase.jl/pull/78/files#diff-a5ad781cd45a87e5f6d2d2c4c6c7fa870e1aa8c50f58bc2f6016f898a20d9b11R20, i.e.,

In the new

 `augment(A::DiffEqArray, sol::AbstractODESolution)`

function in SciMLBase, an error is throw:

```type NILSSForwardSensitivityFunction has no field syms```

(
That function is called when computing the forward sensitivities
https://github.com/SciML/DiffEqSensitivity.jl/blob/8f8753b9fda0300b76a68a22b7f86ee0b7eda635/src/nilss.jl#L245
)

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
@sharanry could you take a look?

**Comment by sharanry** [comment]
@ChrisRackauckas This needs to be reopened again as the last time SciMLBase hadn't been released yet.

**Comment by sharanry** [comment]
@frankschae I don't think the new CI test failures are related to `augment` function?

**Comment by frankschae** [comment]
Thanks a lot, @sharanry !

**Comment by frankschae** [comment]
The multiple shooting error seems to be unrelated/random:
```
Multiple shooting: Test Failed at /home/runner/work/DiffEqSensitivity.jl/DiffEqSensitivity.jl/downstream/test/multiple_shoot.jl:73
  Expression: loss_ms < loss_ss
   Evaluated: 72.87888f0 < 57.684875f0
```
Do we just need a `Random.seed!(seed)` in https://github.com/SciML/DiffEqFlux.jl/blob/master/test/multiple_shoot.jl ?

File: src/nilss.jl [hunk 1]
```diff
@@ -323,11 +323,11 @@ function dudt_g_dgdu!(dudt, gsave, dgdu, nilssprob::NILSSProblem, y, p, iseg)
       end
     else
       if dg_val isa Tuple
-        dg[1](dg_val[1],u,uf.p,nothing,j)
-        copyto!(_dgdu, dg_val[1])
+        dg[1](dg_val[1],u,p,nothing,j)
+        copyto!(_dgdu, -dg_val[1])
```
Comment (ChrisRackauckas) [inline src/nilss.jl:327]: This is actually going to give an allocation. Can you do a short PR to make that `_dgdu .= -dg_val[1]`?

File: src/nilss.jl [hunk 2]
```diff
@@ -323,11 +323,11 @@ function dudt_g_dgdu!(dudt, gsave, dgdu, nilssprob::NILSSProblem, y, p, iseg)
...
       else
-        dg(dg_val,u,uf.p,nothing,j)
-        copyto!(_dgdu, dg_val)
+        dg(dg_val,u,p,nothing,j)
+        copyto!(_dgdu, -dg_val)
```
Comment (ChrisRackauckas) [inline src/nilss.jl:330]: same here.

**Pull request merged**